src/Analyzers/Core/Analyzers/UseObjectInitializer/UseNamedMemberInitializerAnalyzer.cs

```diff
@@ -12,224 +12,239 @@
 using Microsoft.CodeAnalysis.Shared.Extensions;
 using Microsoft.CodeAnalysis.UseCollectionInitializer;
 
-namespace Microsoft.CodeAnalysis.UseObjectInitializer
-{
-    internal class UseNamedMemberInitializerAnalyzer<
+namespace Microsoft.CodeAnalysis.UseObjectInitializer;
+
+internal abstract class AbstractUseNamedMemberInitializerAnalyzer<
```

Comment:
this pattern matches what we do in AbstractUseCOllectionInitializerAnalyzer.  It's verbose in terms of the decl here and the generic constraints, but it makes the typing of all the subclasses nicer.